src/Dialog.cc

```diff
@@ -75,3 +78,153 @@ QQuickItem *Dialog::RootItem() const
+    str += ".\nCheck file permissions.";
+  }
+  else
+    out << config;
+
+  return true;
+}
+
+/////////////////////////////////////////////////
+void Dialog::SetDefaultConfig(const std::string &_config)
+{
+  this->dataPtr->config = _config;
+}
+
+/////////////////////////////////////////////////
+std::string Dialog::ReadAttributeValue(const std::string &_path,
+  const std::string &_attribute) const
+{
+  tinyxml2::XMLDocument doc;
+  std::string value {""};
+  std::string config = "<?xml version=\"1.0\"?>\n\n";
+  tinyxml2::XMLPrinter defaultPrinter;
+  bool configExists{true};
+
+  // Check if the passed in config file exists.
+  // (If the default config path doesn't exist yet, it's expected behavior.
+  // It will be created the first time now.)
+  if (!common::exists(_path))
+  {
+    configExists = false;
+    doc.Parse(this->dataPtr->config.c_str());
+    // Process each dialog
+    for (auto dialogElem = doc.FirstChildElement("dialog");
+      dialogElem != nullptr;
+      dialogElem = dialogElem->NextSiblingElement("dialog"))
+    {
+      if(dialogElem->Attribute("name") == this->objectName().toStdString())
+      {
+        value = dialogElem->Attribute(_attribute.c_str());
+      }
+    }
+  }
+  else
+  {
+    auto success = !doc.LoadFile(_path.c_str());
+    if (!success)
+    {
+      return "";
+    }
+
+    // Process each existing dialog
+    for (auto dialogElem = doc.FirstChildElement("dialog");
+      dialogElem != nullptr;
+      dialogElem = dialogElem->NextSiblingElement("dialog"))
+    {
+      if(dialogElem->Attribute("name") == this->objectName().toStdString())
+      {
+        if (dialogElem->Attribute(_attribute.c_str()))
+          value = dialogElem->Attribute(_attribute.c_str());
+      }
+    }
+
+    // config exists but attribute not there read from default config
+    if (value.empty())
+    {
+      tinyxml2::XMLDocument missingDoc;
+      missingDoc.Parse(this->dataPtr->config.c_str());
+
+      for (auto dialogElem = doc.FirstChildElement("dialog");
+        dialogElem != nullptr;
+        dialogElem = dialogElem->NextSiblingElement("dialog"))
+        {
+          if(dialogElem->Attribute("name") == this->objectName().toStdString())
+          {
+            if(dialogElem->Attribute(_attribute.c_str()))
+              value = dialogElem->Attribute(_attribute.c_str());
+          }
+        }
+      missingDoc.Print(&defaultPrinter);
+    }
+  }
+
+  // Write config file
+  tinyxml2::XMLPrinter printer;
+  doc.Print(&printer);
+
+  // Don't write the xml version decleration if file exists
+  if (configExists){
+    config = "";
+  }
+  igndbg << "Setting Quick start dialog default config." << std::endl;
+
+  config += printer.CStr();
+  config += defaultPrinter.CStr();
+  std::ofstream out(_path.c_str(), std::ios::out);
+  if (!out)
+  {
+    std::string str = "Unable to open file: " + _path;
+    str += ".\nCheck file permissions.";
+    igndbg << str << std::endl;
```

Comment:
This looks like an error

```suggestion
    ignerr << str << std::endl;
```

You could also save a copy and just stream the error directly to `ignerr` without going through an intermediate variable `str`